data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3SinkServiceIT.java

```
@@ -0,0 +1,137 @@
+
+        codec = new JsonCodec();
+        bufferFactory = new InMemoryBufferFactory();
+
+        when(objectKeyOptions.getNamePattern()).thenReturn("my-elb-%{yyyy-MM-dd'T'hh-mm-ss}");
+        when(bucketOptions.getBucketName()).thenReturn(bucketName);
+        when(bucketOptions.getObjectKeyOptions()).thenReturn(objectKeyOptions);
+        when(thresholdOptions.getEventCount()).thenReturn(10);
+        when(thresholdOptions.getMaximumSize()).thenReturn(ByteCount.parse("2mb"));
+        when(thresholdOptions.getEventCollectTimeOut()).thenReturn(Duration.parse("PT3M"));
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getBucketOptions()).thenReturn(bucketOptions);
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of(s3region));
+
+    }
+
+    @Test
+    void verify_flushed_records_into_s3_bucket() {
+
+        int s3ObjectCountBeforeIngest = gets3ObjectCount();
+        S3SinkService s3SinkService = createObjectUnderTest();
+        s3SinkService.output(setEventQueue());
+        int s3ObjectCountAfterIngest = gets3ObjectCount();
+        assertThat(s3ObjectCountAfterIngest, equalTo(s3ObjectCountBeforeIngest + 1));
+    }
+
+    private S3SinkService createObjectUnderTest() {
+        return new S3SinkService(s3SinkConfig, bufferFactory, codec);
+    }
+
+    private int gets3ObjectCount() {
+        int s3ObjectCount = 0;
+        try {
+            ListObjectsRequest listObjects = ListObjectsRequest.builder().bucket(bucketName).build();
+            ListObjectsResponse res = s3Client.listObjects(listObjects);
+            List<S3Object> objects = res.contents();
+            s3ObjectCount = objects.size();
+        } catch (S3Exception e) {
+            LOG.error(e.awsErrorDetails().errorMessage());
+            System.exit(1);
```

This is going to fail the whole test suite and will prevent accurate reporting of failures.

Just let the exception be thrown here. Remove the try-catch block entirely. JUnit will handle it.

---

Removed the try-catch block.